uint32 x
uint32 y
uint32 width
uint32 height
---
int32 errorCode
string errorDescription